src/test/scala/io/moia/scalaHttpClient/AwsRequestSignerTest.scala

```diff
@@ -46,5 +45,36 @@ class AwsRequestSignerTest extends AnyWordSpecLike with Matchers with FutureValu
       for (i <- paramValuePairs)
         result.uri.rawQueryString.getOrElse("").contains(i) shouldBe true
     }
+
+    "add the correct headers to a GET request with some header" in {
+      // Name in AWS test data: get-vanilla-query-order
+      val unsignedRequest: HttpRequest =
+        HttpRequest(uri = Uri(s"https://www.moia.io/"), headers = scala.collection.immutable.Seq(RawHeader("someHeader", "123")))
+
+      val result: HttpRequest = underTest.signRequest(unsignedRequest).futureValue
+      // the difference to the test before is that not only the additional header is present
+      // but also included in the SignedHeaders part of the Authorization header
+      val regex =
+        "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/[0-9]{8}/eu-central-1/execute-api/aws4_request, SignedHeaders=host;someheader;x-amz-date, Signature=.*"
+
+      result.headers.find(_.name() == "Authorization").map(_.value()) should not be empty
+      result.headers
+        .find(_.name() == "Authorization")
+        .map(_.value())
+        .getOrElse("")
+        .matches(regex) shouldBe true
+
+      result.headers.find(_.name() == "someHeader").map(_.value()).getOrElse("") should equal("123")
+    }
+
+    "fail a GET request with an already existing Authorization header" in {
```

Comment:
```suggestion
    "fail a request with an already existing Authorization header" in {
```
The `GET` is neither important, nor being tested 😉